docs/features/compilation-from-portable-pdb.md

```diff
@@ -0,0 +1,346 @@
+* errorendlocation
+* errorlog
+* errorreport
+* fullpaths
+* incremental
+* keycontainer
+* keyfile
+* noconfig
+* nologo
+* nowarn
+* out
+* parallel
+* pathmap
+* preferreduilang
+* recurse
+* refout
+* refonly
+* reportanalyzer
+* ruleset
+* utf8output
+* version
+* warn
+* warnaserror
+
+#### Visual Basic Flags That Can Be Derived From PDB or Assembly
+
+* baseaddress
+* checksumalgorithm
+* debug
+* filealign
+* highentropyva
+* linkresource
+    - Will be represented in metadata reference embedded in pdb
+* main 
+* nostdlib
+    - Will be represented in metadata reference embedded in pdb
+* platform
+* resource
+    - Will be represented in metadata reference embedded in pdb
+* subsystemversion
+* target
+* win32icon
+* win32manifest
+* win32resource
+
+#### Visual Basic Flags Not Included
+
+* bugreport
+* delaysign
+* doc
+* errorreport
+* help
+* keycontainer
+* keyfile
+* libpath
+* moduleassemblyname
+* modulename
+* netcf
+* noconfig
+* nologo
+* nowarn
+* nowin32manifest
+* optimize
+* optioncompare
+* optionexplicit
+* optioninfer
+* out
+* parallel
+* preferreduilang
+* quiet
+* recurse
+* refonly
+* refout
+* rootnamespace
+* ruleset
+* sdkpath
+* utf8output
+* vbruntime
+* verbose
+* warnaserror
+
+#### Compiler Version
+
+This will be a full compiler version, including commit SHA for the build of the compiler. This will be used since compilations across compiler versions are not guaranteed to be the same.
+
+#### Source Encoding
+
+Encoding for source files that are not declared using BOM. This 
+
+#### Metadata References
+
+Symbol server uses a [key](https://github.com/dotnet/symstore/blob/master/docs/specs/SSQP_Key_Conventions.md#pe-timestamp-filesize) computed from the COFF header in the PE image:
+
+Timestamp: 4 byte integer
+Size of image: 4 byte integer
+
+Example:
+
+File name: `example.exe` 
+
+COFF header Timestamp field: `0x542d5742` 
+
+COFF header SizeOfImage field: `0x32000` 
+
+Lookup key: `example.exe/542d574232000/example.exe` 
+
+To fully support metadata references, we'll need to be able to find the exact PE image that was used in the compilation. We'll do this by storing the parts that make up the symbol server key.
+
+We will also store the MVID of a reference since it's a GUID that represents the symbol. This is to future proof the information we will need to look up references.
+
+At this time, we'll only include external references for the compilation. Any other references may be added later in a separate blob.
+
+## PDB Format Additions
+
+#### Metadata References
+
+Metadata references can be easily stored as binary. The binary encoding will be as follows (order matters):
+
+Name: A UTF-8 string (null terminated)
+
+Aliases: UTF-8 Comma (, ) separate list of aliases (null terminated). May be empty
+
+MetadataImageKind: byte value representing Microsoft. CodeAnalysis. MetadataImageKind
+
+EmbedInteropTypes/MetadataImageKind: byte, reads as binary values starting in order from right to left
+
+    
```

Comment:
Double blank line

Reply:
Needed for formatting

This is what the preview looks like in vscode

![image](https://user-images.githubusercontent.com/475144/83439887-cfb23100-a3f8-11ea-913c-2499b1db85ca.png)

Reply:
FWIW you can use the rich diff in GitHub to see diff of what is generated. I just moved a bunch of stuff around to hopefully help with readability and reduce redundant statements